pkg/reconciler/serverlessservice/serverlessservice.go

```diff
@@ -187,17 +187,28 @@ func (r *reconciler) reconcilePublicEndpoints(ctx context.Context, sks *netv1alp
 
 	var (
 		srcEps                *corev1.Endpoints
-		activatorEps          *corev1.Endpoints
-		err                   error
 		foundServingEndpoints bool
 	)
-	activatorEps, err = r.endpointsLister.Endpoints(system.Namespace()).Get(activator.K8sServiceName)
+	activatorEps, err := r.endpointsLister.Endpoints(system.Namespace()).Get(activator.K8sServiceName)
 	if err != nil {
 		logger.Errorw("Error obtaining activator service endpoints", zap.Error(err))
 		return err
 	}
 	logger.Debugf("Activator endpoints: %s", spew.Sprint(activatorEps))
 
+	psn := sks.Status.PrivateServiceName
+	pvtEps, err := r.endpointsLister.Endpoints(sks.Namespace).Get(psn)
+	if err != nil {
+		logger.Errorw(fmt.Sprintf("Error obtaining private service endpoints: %s", psn), zap.Error(err))
+		return err
+	}
+	// We still might be "ready" even if in proxy mode,
+	// if proxy mode is by means of burst capacity handling.
```

Comment:
Does this mean we can now start up Revisions in Proxy mode instead of Serve mode?

Reply:
Interestingly enough not at first, since decider have not been reconciled yet. But with proper config settings that will be the case after the data propagates through the system.

Reply:
Actually let me take it back.
We already did that _if_ by the time SKS reconciles we did not have ready pods (which is usual, but once we get 0second cold starts might stop being true).